### ISSUE TITLE
cannot put an aria-describedby on the h1 of the labelHeaderLevel: '1' for a va-radio

### ISSUE THREAD

**Issue by rhasselle-oddball** [department-of-veterans-affairs/vets-design-system-documentation#2061]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

When using a screen reader, providing an aria-describedby to an h1 will result in reading the h1, then the describedby part (desired behavior). However we cannot apply an aria-describedby to the h1 inside of a web component like va-radio (when using labelHeaderLevel 1), so it only reads the h1.  When you try to apply message-aria-describedby to the web component, it doesn't work properly.

## What I expected to happen

expect to do something like this inside the shadowDom:
```
<h1 aria-describedby="id-of-an-element">
    Label of element
</h1>
```

## Reproducing
- Formation version:
- Device: (e.g. iPhone 8, Macbook)
- Browser: (e.g. Firefox, IE 11)

Steps to reproduce:

1. Use a v3 va-radio button try to apply 
2. Try to set a message-aria-describedby or aria-describedby on the web component of an id of another element with text in it, and see what happens when you use a screen reader such as nvda (windows) or voice over on mac. It only reads the h1, but I expect it to also read the described by reference

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [x] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

**Comment by jamigibbs** [comment]
Hi @rhasselle-oddball! I believe I understand what you're needing but just to clarify, you are expecting a similar option like the `va-checkbox` component which has a `message-aria-describedby` prop that allows you to pass in an optional string (from outside of the web component) that would be read when an element within the component (the header in this case) is focused?

![Screenshot 2023-09-22 at 10 15 34 AM](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/872479/833cb702-af5f-4d87-982d-1ed9bfeaede7)


https://design.va.gov/storybook/?path=/docs/components-va-checkbox--default

**Comment by jamigibbs** [comment]
A follow-up question for you @rhasselle-oddball; is there a reason that you need this only when a header tag is used? You would need to explicitly tell the component (using the `label-header-level` prop) to display the "label" with a header tag. Otherwise just regular text will display in the legend. In other words:

**Markup With header level prop**

```html
<fieldset>
  <legend>
    <h1>My label text</h1>
  </legend>
</fieldset>
```

**Markup Without header prop**

```html
<fieldset>
  <legend>
    My label text
  </legend>
</fieldset>
```

Do you need the additional aria-describedby context for both of those scenarios?

**Comment by caw310** [comment]
This may carry over. Waiting for clarification on this issue.

**Comment by rhasselle-oddball** [comment]
@jamigibbs in our header usability test (https://staging.va.gov/authorization-to-disclose-alternate), if you go to the first page, this shows what we want. (not using web components here, because we couldn't get it to do what we wanted).
Also including @jeana-adhoc she may have some guidance. 
We discussed recently the issue of using a string vs id - if you can point to an id, then that id can contain complex content and still be read out correctly, but a plain string might be the easier solution.
having a meeting with Jeana is really valuable if you need additional information, as she's been doing a lot of research in this area.

**Comment by rhasselle-oddball** [comment]
otherwise I think we would expect a aria to be applied to the input

**Comment by jamigibbs** [comment]
Thanks @rhasselle-oddball!

> We discussed recently the issue of using a string vs id - if you can point to an id, then that id can contain complex content and still be read out correctly, but a plain string might be the easier solution.

We can not reference an ID that is external to the web component's shadow DOM so I think a plain string would be the only option. I'll start working towards that approach and we can continue reviewing/discussing on the PR if needed.

I'm adding a screenshot of that form page for reference (header is focused when page loads)

![Screenshot 2023-09-26 at 10 28 13 AM](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/872479/f9dace19-7408-4d6c-8778-32c34bbde9d6)
